pkg/repo/github/github.go

```diff
@@ -21,82 +21,90 @@ const (
 type gitCrawler struct {
 	owner    string
 	repoName string
-	ref      string
+	tag      string
 }
 
-func NewGitCrawler(owner string, repoName string, ref string) (repo.RepoCrawler, error) {
+func NewGitCrawler(owner string, repoName string, tag string) (repo.RepoCrawler, error) {
 	if owner == "" || repoName == "" {
 		return nil, errors.New("invalid repository identifier")
 	}
 
 	return &gitCrawler{
 		owner:    owner,
 		repoName: repoName,
-		ref:      ref,
+		tag:      tag,
 	}, nil
 }
 
 func (g gitCrawler) DownloadRepo() (string, error) {
-	uri := fmt.Sprintf(fetchRepoURI, g.owner, g.repoName, g.ref)
+	uri := fmt.Sprintf(fetchRepoURI, g.owner, g.repoName, g.tag)
 
 	resp, err := http.Get(uri)
 	if err != nil {
 		return "", err
 	}
 	defer resp.Body.Close()
 
-	tmpSubDir := fmt.Sprintf(tmpFileName, g.owner, g.repoName, g.ref)
-	tmpDir := filepath.Join(os.TempDir(), tmpSubDir)
-	if err := untar(resp.Body, tmpDir); err != nil {
+	tmpDir := os.TempDir()
+	dirName, err := untar(resp.Body, tmpDir)
+	if err != nil {
 		os.Remove(tmpDir)
 		return "", err
 	}
 
-	return tmpDir, nil
+	return dirName, nil
 }
 
-func untar(content io.Reader, tmpDir string) error {
+func untar(content io.Reader, tmpDir string) (string, error) {
 	gzr, err := gzip.NewReader(content)
 	defer gzr.Close()
 	if err != nil {
-		return err
+		return "", err
 	}
 	tr := tar.NewReader(gzr)
+	var dirName string
 
+fileLoop:
 	for {
 		hdr, err := tr.Next()
 		switch {
 		case err == io.EOF:
-			break
+			break fileLoop
```

Comment:
can you try to refactor this to avoid the goto? they always lead to go code that's tough to follow